Dapper/SqlMapper.cs

```diff
@@ -3097,6 +3068,121 @@ private static LocalBuilder GetTempLocal(ILGenerator il, ref Dictionary<Type, Lo
                 throw MultiMapException(reader);
             }
 
+            var returnType = type.IsValueType() ? typeof(object) : type;
+            var dm = new DynamicMethod("Deserialize" + Guid.NewGuid().ToString(), returnType, new[] { typeof(IDataReader) }, type, true);
+            var il = dm.GetILGenerator();
+
+            if (IsValueTuple(type))
+            {
+                GenerateValueTupleDeserializer(type, reader, startBound, length, returnNullIfFirstMissing, il);
+            }
+            else
+            {
+                GenerateDeserializerFromMap(type, reader, startBound, length, returnNullIfFirstMissing, il);
+            }
+
+            var funcType = System.Linq.Expressions.Expression.GetFuncType(typeof(IDataReader), returnType);
+            return (Func<IDataReader, object>)dm.CreateDelegate(funcType);
+        }
+
+        private static void GenerateValueTupleDeserializer(Type valueTupleType, IDataReader reader, int startBound, int length, bool returnNullIfFirstMissing, ILGenerator il)

```

Comment:
@mgravell (For review, before I forget) The parameter `returnNullIfFirstMissing` is unused. Should I be using it, or can I rule that out and remove the parameter since the deserialization type is a value type?

Reply:
yeah, I think you're fine to remove it... but could always just add a test that adds a `(int,int)?` type and zero rows?

Reply:
@mgravell `connection.QuerySingleOrDefault<(int, int)?>("select null where 0 = 1");` does not result in `returnNullIfFirstMissing` becoming true. Should I be trying something with multi-mapping? (Looks like the parameter was added in https://github.com/StackExchange/Dapper/commit/e11257fdf82a4307523e09859e31da3d495877a6)

Reply:
I wrote this test against master to prove that nullable C# tuples have never been supported:

```cs
[Fact]
public void NullableTupleMultiMap()
{
    var outer = connection.Query<Foo, (int, int)?, Foo>(
        "select * from (values (42, null, null), (42, 43, 44)) as Foo (A, B1, B2)",
        (foo, b) =>
        {
            foo.B = b;
            return foo;
        },
        splitOn: "B1");

    Assert.Equal(42, outer.First().A);
    Assert.Null(outer.First().B);

    Assert.Equal(42, outer.Last().A);
    Assert.Equal(43, outer.Last().B.Value.Item1);
    Assert.Equal(44, outer.Last().B.Value.Item2);
}

private sealed class Foo
{
    public int A { get; set; }
    public (int, int)? B { get; set; }
}
```

And therefore `returnNullIfFirstMissing` is not meaningful yet and I can remove it.